torch/csrc/autograd/utils/grad_layout_contract.h

```diff
@@ -13,6 +13,8 @@ namespace utils {
 inline bool obeys_layout_contract(const at::Tensor& grad, const at::Tensor& variable) {
   TORCH_INTERNAL_ASSERT(!grad.is_sparse());
   TORCH_INTERNAL_ASSERT(!variable.is_sparse());
+  TORCH_INTERNAL_ASSERT(!grad.is_sparse_csr());
```

Comment:
We really need a `is_any_sparse()` :p

Reply:
@albanD How about ```torch.Tensor.has_layout(Layout[])```?

Reply:
And when more layouts are being added soon we do really need `is_any_sparse()`. I like cpuhrsch's suggestion to check whether the layout matches the given list of layouts.